File: javascript/rna-transcription/rna_transcription.js

var TRANSCRIPTIONS = {
  C: 'G',
  G: 'C',
  A: 'U',
  T: 'A'
}

function withTheirTranscriptions(nucleotide) {
  return TRANSCRIPTIONS[nucleotide];
}

module.exports = function toRna(dna) {
  return dna.replace(/[CGAT]/g, withTheirTranscriptions);
}

File: elixir/rna-transcription/rna_transcription.exs
defmodule RNATranscription do

  @dna_to_rna %{
    ?G => ?C,
    ?C => ?G,
    ?T => ?A,
    ?A => ?U
  }

  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna |> Enum.map(fn nucleotide -> @dna_to_rna[nucleotide] end)
  end

end
